src/Documentation/Action/DocumentationAction.php

```diff
@@ -47,6 +67,12 @@ public function __invoke(Request $request = null): Documentation
                 $context['api_gateway'] = true;
             }
             $request->attributes->set('_api_normalization_context', $request->attributes->get('_api_normalization_context', []) + $context);
+
+            $attributes = RequestAttributesExtractor::extractAttributes($request);
+        }
+        // BC check to be removed in 3.0
+        if (null !== $this->formatsProvider) {
+            $this->formats = $this->formatsProvider->getFormatsFromAttributes($attributes ?? []);
```

Comment:
There is no need to declare `$attributes` (line 71) out of this `if`, it could be:
```php
$this->formats = $this->formatsProvider->getFormatsFromAttributes(RequestAttributesExtractor::extractAttributes($request));
```